### DISCUSSION TITLE
Codespaces GPG signing fails in instances other than `git commit -S`, like `git tag -s` or `git cherry-pick`

### DISCUSSION THREAD

**Discussion by possumbilities** [community/community#33947] [Codespaces]
It appears that signing with GPG keys and their availability in the Codespace is severely limited to ONLY 
`git commit -S -m 'message'`, other operations seem to fail.

For example: 

* `git tag -s tag `
* `git cherry-pick hash`

Both return an error that gpg failed.

However, 

`git commit -S -m 'message'` successfully signs

See this related issue here, that just covers tag signing failure: #20990

**Comment by JacobsenKim** [comment]
i have same problem with tag
but if i run this

`git config --global gpg.program /usr/bin/gpg`

its working but now it use the <noreply@github.com>

if i go back to 

`git config --global gpg.program /.codespaces/bin/gh-gpgsign`

it fail the tag

i cant figure out to edit the info on `git config --global gpg.program /usr/bin/gpg` so its not having the verify sign on github page becuse off the wrong email.



I know there is not setup a key on this codespace yet, its 100% fresh but its shows the email.
![Skærmbillede fra 2024-01-31 05-34-45](https://github.com/community/community/assets/157890151/d4b1fd56-0c72-418b-a6c9-0b927b4a825e)
